pfcpiface/pfcpiface.go

```diff
@@ -352,3 +380,51 @@ func handleSessionDeletionRequest(upf *upf, msg message.Message, addr net.Addr,
 
 	return smres
 }
+
+func initiateAssociationSetupRequest(sourceIP string, n3ip string, n4DstIp string, conn *net.UDPConn, ch chan bool) {
+	c1 := make(chan int, 1)
+	numRetransmissions := 0
+	for {
+		// Build request message
+		asreq, err := message.NewAssociationSetupRequest(ie.NewRecoveryTimeStamp(time.Now()),
+			ie.NewNodeID(sourceIP, "", ""), /* node id (IPv4) */
+			// 0x41 = Spare (0) | Assoc Src Inst (1) | Assoc Net Inst (0) | Tied Range (000) | IPV6 (0) | IPV4 (1)
+			//      = 01000001
+			ie.NewUserPlaneIPResourceInformation(0x41, 0, n3ip, "", "", ie.SrcInterfaceAccess),
+		).Marshal() /* userplane ip resource info */
+		if err != nil {
+			log.Fatalln("Unable to create association setup response", err)
+		}
+
+		smfAddr, err := net.ResolveUDPAddr("udp", n4DstIp+":"+PFCPPort)
+		if err != nil {
+			log.Fatalln("Unable to resolve udp addr!", err)
+			return
+		}
+
+		if _, err := conn.WriteTo(asreq, smfAddr); err != nil {
+			log.Fatalln("Unable to transmit association setup request ", err)
+		}
+		go func() {
+			time.Sleep(2 * time.Second)
+			c1 <- 1
+		}()
```

Comment:
this goroutine would leak right? I dont see how it exits

Reply:
Can we replace with ticker or timer?

Reply:
Isn't it like it go routine just write on the channel once and returns. If pfcp-agent receives the response from the control plane before timeout then the function initiateAssociationSetupRequest returns and channel would get destroyed.. So I don't see why this will cause any leak.